compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -5052,11 +5052,12 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
 
         Linter.warnOnImplausiblePattern(tree, pt)
 
-        val cmp =
-          untpd.Apply(
-            untpd.Select(untpd.TypedSplice(tree), nme.EQ),
-            untpd.TypedSplice(dummyTreeOfType(pt)))
-        typedExpr(cmp, defn.BooleanType)
+        if ! (tree.tpe <:< pt && (tree.symbol.flags.isAllOf(Flags.EnumValue) || (tree.symbol.flags.isAllOf(Flags.Module | Flags.Case)))) then
```

Comment:
Some remarks:

 1. The motivation for the SIP was that it makes strictEquality easier to use, but this also changes the pattern matching behavior of patterns without that import. And it breaks the law that pattern matching values is done with an `==` check. I think this goes too far, even though this is what's specified in the SIP. 

 2. We cannot implement a SIP change the behavior of the compiler at large without going through an experimental language import. 
 
 3. I think it would be better to add code to `assumedCanEqual` instead. Under strictEquality, if the left operand (or either one) is an enum constant, and the other operand is a supertype, assume the two `CanEqual`.